middleware_timeout.go

```diff
@@ -10,7 +10,7 @@ import (
 )
 
 // Timeout creates middleware to set a timeout for a request
-func Timeout(timeout time.Duration) contractshttp.Middleware {
+func Timeout(timeout time.Duration, recoverFunc func(ctx contractshttp.Context, err interface{})) contractshttp.Middleware {
```

Comment:
The `recoverFunc` method doesn't need to be passed here, use the global `recoverFunc` directly.